community/kernel/src/main/java/org/neo4j/kernel/api/security/AuthSubject.java

```diff
@@ -54,6 +54,22 @@
     String username();
 
     /**
+     * @param username a username
+     * @return true if the provided username is the underlying user name of this subject
+     */
+    boolean hasUsername( String username );
+
+    /**
+     * Ensure that the provided username is the name of an existing user known to the system.
+     *
+     * @param username a username
+     * @throws InvalidArgumentsException if the provided user name is not the name of an existing user
+     */
+    default void ensureUserExistsWithName( String username ) throws InvalidArgumentsException {
```

Comment:
I find 2 strange things with thin method here: 
- i think ensureUserExistsWithName should not be part of AuthSubject interface at all since it does not look like that AuthSubject should be responsible for creation of users
- since it's our first implementation of AuthSubject interface i believe that we should not declare it `default`. If we need some default behaviour it should be an abstract class instead